components/Buttons/src/MDCFloatingButton.m

```diff
@@ -71,55 +112,171 @@ - (instancetype)initWithFrame:(CGRect)frame shape:(MDCFloatingButtonShape)shape
   self.layer.cornerRadius = CGRectGetHeight(self.bounds) / 2;
   [super layoutSubviews];
+
+   if (self.mode == MDCFloatingButtonModeNormal) {
+    return;
+  }
+
+  // Position the imageView and titleView
+  //
+  // +------------------------------------+
+  // |    |  |  |  CEI TOP            |   |
+  // |CEI +--+  |       +-----+       |CEI|
+  // | LT ||SP|-- A --|Title|-- A --|RGT|
+  // |    +--+  |       +-----+       |   |
+  // |    |  |  |  CEI BOT            |   |
+  // +------------------------------------+
+  //
+  // (A) The same spacing on either side of the label.
+  // (SP) The spacing between the image and title
+  // (CEI) Content Edge Insets
+  //
+  // The diagram above assumes an LTR user interface orientation
+  // and a .leadingIcon imageLocation for this button.
+
+  BOOL isLeadingIcon = self.imageLocation == MDCFloatingButtonImageLocationLeading;
+  UIEdgeInsets adjustedLayoutInsets =
+      isLeadingIcon ? internalLayoutSpacingInsets
+                    : UIEdgeInsetsFlippedHorizonally(internalLayoutSpacingInsets);
+
+  const CGRect insetBounds = UIEdgeInsetsInsetRect(UIEdgeInsetsInsetRect(self.bounds,
+                                                                         adjustedLayoutInsets),
+                                                   self.contentEdgeInsets);
+
+  const CGFloat imageViewWidth = CGRectGetWidth(self.imageView.bounds);
+  const CGFloat boundsCenterY = CGRectGetMidY(insetBounds);
+  CGFloat titleWidthAvailable = CGRectGetWidth(insetBounds);
+  titleWidthAvailable -= imageViewWidth;
+  titleWidthAvailable -= self.imageTitleSpacing;
+
+  const CGFloat availableHeight = CGRectGetHeight(insetBounds);
+  CGSize titleIntrinsicSize
+      = [self.titleLabel sizeThatFits:CGSizeMake(titleWidthAvailable, availableHeight)];
+
+  const CGSize titleSize = CGSizeMake(MAX(0, MIN(titleIntrinsicSize.width, titleWidthAvailable)),
+                                      MAX(0, MIN(titleIntrinsicSize.height, availableHeight)));
+
+  CGPoint titleCenter;
+  CGPoint imageCenter;
+  BOOL isRTL = self.mdf_effectiveUserInterfaceLayoutDirection
+      == UIUserInterfaceLayoutDirectionRightToLeft;
+
+  // isRTL && isLeadingIcon => icon to right of text
+  // !isRTL && !isLeadingIcon => icon to right of text
+  // isRTL && !isLeadingIcon => icon to left of text
+  // !isRTL && isLeadingIcon => icon to left of text
+  if (isRTL == isLeadingIcon) {
+    const CGFloat imageCenterX
+        = CGRectGetMaxX(insetBounds) - (imageViewWidth / 2);
+    const CGFloat titleCenterX = CGRectGetMinX(insetBounds) + (titleWidthAvailable / 2);
+    imageCenter = CGPointMake(imageCenterX, boundsCenterY);
+    titleCenter = CGPointMake(titleCenterX, boundsCenterY);
+  } else {
+    const CGFloat imageCenterX
+        = CGRectGetMinX(insetBounds) + (imageViewWidth / 2);
+    const CGFloat titleCenterX
+        = CGRectGetMaxX(insetBounds) - (titleWidthAvailable / 2);
```

Comment:
Nit: These assignments don't need to wrap multiple lines.

Reply:
Done